server.go

```diff
@@ -67,6 +67,9 @@ type server struct {
 	config  *Config
 
 	conn net.PacketConn
+	// If the server is started with ListenAddr, we create a packet conn.
+	// If it is started with Listen, we take a packet conn as a parameter.
+	createdPacketConn bool
```

Comment:
Nit: Maybe shouldCloseConn?

Reply:
We have the same in the client, so I'd prefer to leave it like this.